plugins/scheduled_task/lib/object_task_engine/KObjectTaskModifyEntryEngine.php

```diff
@@ -0,0 +1,158 @@
+
+		try
+		{
+			$client->baseEntry->update($entryId, $entryObj);
+		}
+		catch(Exception $e)
+		{
+			KalturaLog::notice("problem with entry update entry id $entryId - " . $e->getMessage());
+		}
+		KBatchBase::unimpersonate();
+	}
+	
+	private static function updateMetadataObj(KalturaBaseEntry $entryResultForMetadataUpdate, &$metadataPlugin, $outputMetadataArr, $outputMetadataProfileId, $metadataFilter)
+	{
+		$entryId = $entryResultForMetadataUpdate->id;
+		
+		try
+		{
+			$metadataInputResult = $metadataPlugin->metadata->listAction($metadataFilter, null);
+		}
+		catch(Exception $e)
+		{
+			KalturaLog::notice("problem with metadata list with entry id $entryId - " . $e->getMessage());
+		}
+	
+		if($metadataInputResult && $metadataInputResult->totalCount != 0)
+		{
+			$metadataId = $metadataInputResult->objects[0]->id;
+			$metadataInputXmlStr = $metadataInputResult->objects[0]->xml;
+			$xmlObj = new SimpleXMLElement($metadataInputXmlStr);
+			
+			$xmlData = self::insertMetadataValuesToEntryObj($entryResultForMetadataUpdate, $xmlObj, $outputMetadataArr);
+			
+			try
+			{
+				$metadataPlugin->metadata->update($metadataId, $xmlData, null);
+			}
+			catch(Exception $e)
+			{
+				KalturaLog::notice("problem with metadata update entry id $entryId - " . $e->getMessage());
+			}
+		}
+		else
+		{
+			$xmlObj = new SimpleXMLElement("<metadata></metadata>");
+			$xmlData = self::insertMetadataValuesToEntryObj($entryResultForMetadataUpdate, $xmlObj, $outputMetadataArr);
+	
+			try
+			{
+				$metadataPlugin->metadata->add($outputMetadataProfileId, KalturaMetadataObjectType::ENTRY, $entryId, $xmlData);
+			}
+			catch(Exception $e)
+			{
+				KalturaLog::notice("problem with metadata add entry id $entryId - " . $e->getMessage());
+			}
+		}
+	}
+	
+	private static function insertMetadataValuesToEntryObj(KalturaBaseEntry $entryResultForMetadataUpdate, SimpleXMLElement $xmlObj, array $outputMetadataArr)
+	{
+		foreach($outputMetadataArr as $outputMetadataItem)
+		{
+			$xpathName = $outputMetadataItem->key;
+			$fieldName = $outputMetadataItem->value;
+			$xmlObj->$xpathName = $entryResultForMetadataUpdate->$fieldName;
+		}
+	
+		return $xmlObj->asXML();
+	}
+	
+	private static function updateEntryFromMetadata(&$metadataPlugin, array $inputMetadataArr, KalturaBaseEntry $entryObj, $metadataFilter)
+	{
+		$entryId = $entryObj->id;
+		
+		try
+		{
+			$metadataInputResult = $metadataPlugin->metadata->listAction($metadataFilter, null);
+		}
+		catch(Exception $e)
+		{
+			KalturaLog::notice("problem with metadata list - " . $e->getMessage());
+		}
+	
+		if($metadataInputResult->totalCount != 0)
+		{
+			$metadataInputXmlStr = $metadataInputResult->objects[0]->xml;
+			$xmlObj = new SimpleXMLElement($metadataInputXmlStr);
+			
+			foreach($inputMetadataArr as $inputMetadataItem)
+			{					
+				$xpathName = $inputMetadataItem->key;
+				$fieldName = $inputMetadataItem->value;
+	
+				$metadataValue = $xmlObj->$xpathName;
```

Comment:
To better explain my comment that was removed by last commit.
Lets assume that we have the following metadata on the entry:
```xml
<metadata>
<FieldA>ValueInField1</FieldA>
<FieldC>ValueInField3</FieldC>
</metadata>
```

We want to append `FieldB`.
The XSD defines the following order:
- FieldA
- FieldB
- FieldC

Current code will produce and try to save:
```xml
<metadata>
<FieldA>ValueInField1</FieldA>
<FieldC>ValueInField3</FieldC>
<FieldB>ValueInField2</FieldB>
</metadata>
```
This will not pass XSD validation.